spec/models/manageiq/providers/vmware/cloud_manager/vm_spec.rb

```diff
@@ -50,4 +50,33 @@
       vm.raw_destroy
     end
   end
+
+  describe 'power operations' do
+    before(:each) do
+      allow(ems).to receive(:with_provider_connection).and_yield(connection)
+    end
+
+    let(:ems) { FactoryGirl.create(:ems_vmware_cloud) }
+    let(:vm) { FactoryGirl.create(:vm_vcloud, :ext_management_system => ems, :ems_ref => 'id') }
+    let(:connection) { double('connection') }
+    let(:response) { double('response', :body => nil) }
+
+    context '.raw_stop' do
+      it 'stops the virtual machine' do
+        expect(connection).to receive(:post_undeploy_vapp).with('id', :UndeployPowerAction => 'powerOff').and_return(response)
+        expect(connection).to receive(:process_task).and_return(true)
```

Comment:
You don't really need the `.and_return(true)` part, please feel free to remove it (also below, for suspend).